xarray/core/utils.py

```diff
@@ -805,6 +809,48 @@ def drop_dims_from_indexers(
         )
 
 
+def drop_missing_dims(
```

Comment:
This looks good. We could reduce the code reuse a bit, with only a performance penalty if an invalid `missing_dims` is passed. But it's also fine as-is.